test/suite/test_checkpoint07.py

```diff
@@ -0,0 +1,192 @@
+        # two and table three during the checkpoint. If they're different check
+        # they are within 1 second of each other.
+        self.assertEqual(val2, val3)
+
+        # Now force a checkpoint on clean tables. No clean timer should be set.
+        self.session.checkpoint('force=true')
+        val = self.get_stat(self.uri1)
+        self.assertEqual(val, 0)
+        val = self.get_stat(self.uri2)
+        self.assertEqual(val, 0)
+        val = self.get_stat(self.uri3)
+        self.assertEqual(val, 0)
+
+        # Modify the first two tables and reverify all three.
+        c1[3] = 3
+        c2[3] = 3
+        self.session.checkpoint(None)
+        val = self.get_stat(self.uri1)
+        self.assertEqual(val, 0)
+        val = self.get_stat(self.uri2)
+        self.assertEqual(val, 0)
+        val = self.get_stat(self.uri3)
+        self.assertNotEqual(val, 0)
+
+        # Open a backup cursor. This will pin the most recent checkpoint.
+        # Modify table 1 and checkpoint, then modify table 2 and checkpoint.
+        # The open backup cursor will cause table 1 to get the smaller timer.
+        backup_cursor = self.session.open_cursor('backup:', None, None)
+        c1[4] = 4
+        self.session.checkpoint(None)
+
+        val = self.get_stat(self.uri1)
+        self.assertEqual(val, 0)
+
+        # Now that we have a cursor pinning checkpoints on table 1, modify
+        # table 2 so that the clean setting gets set on table 1. The stat
+        # value should be a smaller timer than table 3 but it should be set.
+        c2[4] = 4
+        self.session.checkpoint(None)
+        val2 = self.get_stat(self.uri2)
+        self.assertEqual(val2, 0)
+
+        val1 = self.get_stat(self.uri1)
+        val3 = self.get_stat(self.uri3)
+        # Assert table 1 does not have the forever timer value, but it is set.
+        # This assumes table 3 has the forever value.
+        self.assertNotEqual(val1, 0)
+        self.assertNotEqual(val3, 0)
+        self.assertLess(val1, val3)
+        # Save the old forever value from table 3.
+        oldval3 = val3
+
+        # Force a checkpoint while the backup cursor is open. Then write again
+        # to table 2. Since table 1 and table 3 are clean again, this should
+        # force both table 1 and table 3 to have the smaller timer.
+        self.session.checkpoint('force=true')
+        val1 = self.get_stat(self.uri1)
+        val3 = self.get_stat(self.uri3)
+        self.assertEqual(val1, 0)
+        self.assertEqual(val3, 0)
+        c2[5] = 5
+        self.session.checkpoint(None)
+        val2 = self.get_stat(self.uri2)
+        self.assertEqual(val2, 0)
+
+        val1 = self.get_stat(self.uri1)
+        val3 = self.get_stat(self.uri3)
+        self.assertNotEqual(val1, 0)
+        self.assertNotEqual(val3, 0)
+        # It is possible that we could span the second timer when processing table
+        # two and table three during the checkpoint. If they're different check
```

Comment:
As above, the off-by-one check seems to be missing.  Also, I assume this comment got truncated in the copy.

Reply:
Fixed